gunrock/app/lp/lp_problem.cuh

```diff
@@ -0,0 +1,511 @@
+
+  typedef ProblemBase<GraphT, FLAG> BaseProblem;
+  typedef DataSliceBase<GraphT, FLAG> BaseDataSlice;
+
+  // Helper structures
+  /**
+   * @brief Data slice structure containing BFS-specific data on indiviual GPU
+   */
+  struct DataSlice : BaseDataSlice {
+    // TODO remove unneccesary fields from here
+    // util::Array1D<SizeT, VertexT> original_vertex;
+    util::Array1D<SizeT, LabelT>
+        labels;  // labels to mark latest iteration the vertex been visited
+    util::Array1D<SizeT, LabelT>
+    old_labels;
+    util::Array1D<SizeT, VertexT> preds;       // predecessors of vertices
+    util::Array1D<SizeT, VertexT> temp_preds;  // predecessors of vertices
+    util::Array1D<SizeT, SizeT> vertex_markers[2];
+    util::Array1D<SizeT, VertexT> unvisited_vertices[2];
+    util::Array1D<SizeT, SizeT, util::PINNED> split_lengths;
+    util::Array1D<SizeT, VertexT> local_vertices;
+    util::Array1D<SizeT, MaskT> visited_masks;
+    util::Array1D<SizeT, MaskT> old_mask;
+    util::Array1D<SizeT, MaskT *> in_masks;
+    // need two arrays
+    // store data array containing labels
+    // store segment information
+    // store the count of segments
+    // max segment size is frontier size
+    // max data array is sum of top frontier size neighbours lengths
+    util::Array1D<SizeT, LabelT> data;
+    util::Array1D<SizeT, int> segments;
+    util::Array1D<SizeT, int> segments_temp;
+    util::Array1D<uint64_t, char> cub_temp_storage;
+    // do I have to make these an array?
+    // they are just one variable but I need to call atomicAdd on them
+    util::Array1D<SizeT, int> data_size;
+    util::Array1D<SizeT, int> segments_size;
+    // SizeT data_size, segments_size;
+    SizeT num_visited_vertices, num_unvisited_vertices;
+    bool been_in_backward;
+
+    /*
+     * @brief Default constructor
+     */
+    DataSlice() : BaseDataSlice() {
+      // original_vertex        .SetName("original_vertex"      );
+      labels.SetName("labels");
+      old_labels.SetName("old_labels");
+
+      preds.SetName("preds");
+      temp_preds.SetName("temp_preds");
+      vertex_markers[0].SetName("vertex_markers[0]");
+      vertex_markers[1].SetName("vertex_markers[1]");
+      unvisited_vertices[0].SetName("unvisited_vertices[0]");
+      unvisited_vertices[1].SetName("unvisited_vertices[1]");
+      local_vertices.SetName("local_vertices");
+      split_lengths.SetName("split_length");
+      visited_masks.SetName("visited_masks");
+      old_mask.SetName("old_mask");
+      in_masks.SetName("in_masks");
+  
+      data.SetName("data");
+      data.SetName("data_size");
+
+      segments.SetName("segments");
+      segments.SetName("segments_size");
+      cub_temp_storage.SetName("cub_temp_storage");
+      segments_temp.SetName("segments_temp");
+    }
+
+    /*
+     * @brief Default destructor
+     */
+    virtual ~DataSlice() { Release(); }
+
+    cudaError_t Release(util::Location target = util::LOCATION_ALL) {
```

Comment:
Have you run your program with cuda-memcheck, both before and after your changes? It's a good way to check whether you're releasing all your resources properly.